webpack.js

```diff
@@ -85,18 +89,33 @@ const config = {
 			}],
 		}),
 	],
-}
+})
 
 // Remove svg from default url-loader
 const svgRule = webpackConfig.module.rules.find(rule => rule.test && rule.test.toString().indexOf('|svg') !== -1)
 svgRule.test = new RegExp(svgRule.test.toString().replace('|svg', ''))
 
 // Merge configs
-const mergedConfigs = merge(config, webpackConfig)
+const mainAppMergedConfig = merge(mainAppConfig, webpackConfig)
 
 // Remove duplicate rules by the `test` key
-mergedConfigs.module.rules = mergedConfigs.module.rules
+mainAppMergedConfig.module.rules = mainAppMergedConfig.module.rules
 	.filter((v, i, a) => a.findIndex(t => (t.test.toString() === v.test.toString())) === i)
 
+mainAppMergedConfig.name = 'mainApp'
+mainAppMergedConfig.mode = process.env.NODE_ENV
+mainAppMergedConfig.entry = './src/main.js'
+
+const filesAppIntegrationConfig = Object.assign({}, basicConfig, {
+	name: 'filesAppIntegration',
+	mode: process.env.NODE_ENV,
+	entry: './src/filesbutton.js',
+	output: {
+		path: path.join(__dirname, '/js'),
+		filename: 'filesbutton.min.js',
+	},
+	devtool: 'source-map', // Avoid CSP issues with our files app hook script
+})
```

Comment:
That is not needed! Everything is already handled by the webpack base config.

All you need to do on this file is add something like. Then it will be compiled automatically with the appropriate loaders. 

```js
webpackConfig.entry.filesbutton = path.resolve(path.join('src', 'filesbutton.js'))
```

Reply:
Thanks, done